tests/core/async_module/test_async_module.py

```diff
@@ -0,0 +1,66 @@
+import asyncio
+from inspect import iscoroutine
+import pytest
+from web3 import Web3
+from web3._utils.method_callers import (
+    async_caller,
+    CallBinder,
+)
+from web3._utils.toolz import (
+    curry,
+    identity,
+)
+from web3.module import Module
+
+
+async def async_dummy_request_function(method, params):
+    await asyncio.sleep(0.1)
+    return {'result': [method, params]}
+
+
+def blocking_dummy_request_function(method, params):
+    return {'result': (method, params)}
+
+
+class DummyRequestManager:
+    def __init__(*args, **kwargs):
+        pass
+    async def request_async(self, method, params):
+        return await async_dummy_request_function(method, params)
+    def request_blocking(self, method, params):
+        return blocking_dummy_request_function(method, params)
+
+
+@pytest.fixture()
+def w3():
+    DummyWeb3 = type(
+        'TestWeb3',
+        (Web3,),
+        {'RequestManager': DummyRequestManager})
+    w3 = DummyWeb3()
```

Comment:
Could also try `DummyWeb3(modules=[BlockingModule, AsyncModule])` here instead of attaching below, if you prefer. I haven't used it, but looks like it should theoretically work.